### PR TITLE
Mapper rewrites

### PR DESCRIPTION
**Pull request by Zemurin**
- governmentMapper
- minorityPopMapper
- flagColorMapper

Related files patched accordingly.

### CODE REVIEW

File: EU4toV2/Source/V2World/V2World.cpp [hunk 1]
```diff
@@ -237,32 +238,49 @@ void V2World::importDefaultPops()
 
 	set<string> filenames;
 	Utils::GetAllFilesInFolder("./blankMod/output/history/pops/1836.1.1/", filenames);
+
+	LOG(LogLevel::Info) << "Parsing minority pops mappings";
+
+	std::ifstream minPopFile("minorityPops.txt");
+	if (minPopFile.fail())
+	{
+		LOG(LogLevel::Error) << "Could not parse file minorityPops.txt";
```
Comment (Idhrendur) [inline EU4toV2/Source/V2World/V2World.cpp]: Unless you catch the exception yourself, the top-level try/catch block should log it. So you don't need this line.

File: EU4toV2/Source/V2World/V2World.cpp [hunk 2]
```diff
@@ -539,6 +558,18 @@ void V2World::initializeCountries(const EU4::world& sourceWorld, const mappers::
 	Vic2::TechSchoolsFile theTechSchoolsFile(theBlockedTechSchoolsFile.takeBlockedTechSchools());
 	auto theTechSchools = theTechSchoolsFile.takeTechSchools();
 
+	LOG(LogLevel::Info) << "Parsing governments mappings";
+
+	std::ifstream governmentMapFile("governmentMapping.txt");
+	if (governmentMapFile.fail())
+	{
+		LOG(LogLevel::Error) << "Could not parse file governmentMapping.txt";
```
Comment (Idhrendur) [inline EU4toV2/Source/V2World/V2World.cpp]: Ditto.

File: EU4toV2/Source/V2World/V2Flags.cpp [hunk 3]
```diff
@@ -368,11 +346,24 @@ void V2Flags::copyFlags() const
...
 {
-	string baseFlagFolder = "flags";
+	std::string baseFlagFolder = "flags";
+
+	LOG(LogLevel::Info) << "Parsing EU4 flag colours";
+	std::string colorFileStr = theConfiguration.getEU4Path() + "/common/custom_country_colors/00_custom_country_colors.txt";
+	std::ifstream colorFile(colorFileStr);
+	if (colorFile.fail())
+	{
+		LOG(LogLevel::Error) << "Could not open file " << colorFileStr;
```
Comment (Idhrendur) [inline EU4toV2/Source/V2World/V2Flags.cpp]: Ditto

File: EU4toV2/Source/Mappers/MinorityPop.cpp [hunk 4]
```diff
@@ -0,0 +1,21 @@
+#include "MinorityPop.h"
+#include "Log.h"
```
Comment (Idhrendur) [inline EU4toV2/Source/Mappers/MinorityPop.cpp]: No logging, so probably don't need the header.

File: EU4toV2/Source/Mappers/MinorityPop.h [hunk 5]
```diff
@@ -0,0 +1,23 @@
+#ifndef MINORITY_POP_H
+#define MINORITY_POP_H
+
+#include <memory>
```
[the remark this answers is not in the document]
Comment (Zemurin) [reply EU4toV2/Source/Mappers/MinorityPop.h]: dropped memory from several files where unneeded.

File: EU4toV2/Source/Mappers/FlagColorMapper.h [hunk 6]
```diff
@@ -1,74 +1,26 @@
...
-			return getInstance()->GetFlagColor(index);
-		}
-
-		static int getNumColors()
-		{
-			return getInstance()->flagColorMapping.size();
-		}
+		FlagColorMapper(std::istream& theStream);
+		int getNumFlagColors() const { return flagColorMapping.size(); }
+		commonItems::Color getFlagColorByIndex(int index) const { return flagColorMapping[index]; }
```
[the remark this answers is not in the document]
Comment (Zemurin) [reply EU4toV2/Source/Mappers/FlagColorMapper.h]: done.

File: EU4toV2/Source/Mappers/FlagColorMapper.h [hunk 7]
```diff
@@ -1,74 +1,26 @@
...
-
-
-
 #ifndef FLAG_COLOR_MAPPER_H_
 #define FLAG_COLOR_MAPPER_H_
 
-
-
-#include "../FlagUtils.h"
 #include <memory>
```
[the remark this answers is not in the document]
Comment (Zemurin) [reply EU4toV2/Source/Mappers/FlagColorMapper.h]: dropped

File: EU4toV2/Source/Mappers/GovernmentMapper.h [hunk 8]
```diff
@@ -1,71 +1,22 @@
...
-
-
-
 #ifndef GOVERNMENT_MAPPER_H
 #define GOVERNMENT_MAPPER_H
 
-
-
 #include <map>
 #include <memory>
```
[the remark this answers is not in the document]
Comment (Zemurin) [reply EU4toV2/Source/Mappers/GovernmentMapper.h]: dropped

File: EU4toV2/Source/V2World/V2World.cpp (hunk 1 above)
Comment (Zemurin) [reply EU4toV2/Source/V2World/V2World.cpp]: dropped.

File: EU4toV2/Source/V2World/V2World.cpp (hunk 2 above)
Comment (Zemurin) [reply EU4toV2/Source/V2World/V2World.cpp]: dropped

File: EU4toV2/Source/V2World/V2Flags.cpp (hunk 3 above)
Comment (Zemurin) [reply EU4toV2/Source/V2World/V2Flags.cpp]: dropped

File: EU4toV2/Source/Mappers/MinorityPop.cpp (hunk 4 above)
Comment (Zemurin) [reply EU4toV2/Source/Mappers/MinorityPop.cpp]: dropped

**Pull request merged**